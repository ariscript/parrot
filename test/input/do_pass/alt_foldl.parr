let rec foldl = (lambda (f, acc, l):
    if l == nil:
        acc
    else:
        let (first, rest) = l in 
        foldl(f, f(acc, first), rest))
in
let mylist = (1, (2, (3, nil))) in
let nest = (lambda (a, b): (a, b)) in
foldl(nest, nil, mylist) 
