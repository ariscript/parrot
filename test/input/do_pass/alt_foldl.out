(((nil, 1), 2), 3)
